src/util/symbol_table.cpp

```diff
@@ -128,6 +128,10 @@ void symbol_tablet::validate(const validation_modet vm) const
     const auto symbol_key = elem.first;
     const auto &symbol = elem.second;
 
+    // First of all, ensure symbol well-formedness
+    DATA_CHECK_WITH_DIAGNOSTICS(
```

Comment:
I'd perhaps suggest doing this DATA_CHECK inside symbolt rather than here, and instead just call `symbol.validate(vm)` here..

Reply:
I don't like this that much. I implemented it, but I think I prefer it the way it was before. It makes the logic in `is_well_formed` to appear simpler, and be very easily extendable, and just adds an extra invariant in the sequence of invariants inside `symbol_tablet::validate`, which also makes sense, as it reads, for every symbol, check that they symbol is well formed, then ...